app/views/modals/set-home-page-modal.html

```diff
@@ -0,0 +1,50 @@
+<div class="modal-resource-action">
+  <div class="modal-header">
+    <h1>Set Home Page</h1>
+  </div>
+  <div class="modal-body indent">
+    <p>
+      <div class="title">Set your preferred home page.</div>
+      Note: This setting is browser specific and will not be maintained across browsers.
+    </p>
+    <ng-form name="form">
+      <fieldset ng-disabled="loading">
+        <div class="radio">
+          <label>
+            <input type="radio" ng-model="preferredHomePage" value="catalog-home">
+            Catalog Home (Default)
+          </label>
+        </div>
+        <div class="radio">
+          <label>
+            <input type="radio" ng-model="preferredHomePage" value="project-list">
+            My Projects List
+          </label>
+        </div>
+        <div class="radio" ng-show="availableProjects.length >= 1">
+          <label>
+            <input type="radio" ng-model="preferredHomePage" value="project-overview">
+            Overview Page for {{availableProjects.length === 1 ? (selectedProject | displayName) : ':'}}
+          </label>
+          <div class="select-project-container">
+            <select-project ng-if="availableProjects.length > 1"
+                            class="select-project"
+                            on-project-selected="onProjectSelected"
+                            on-open="onOpen"
+                            available-projects="availableProjects"
+                            selected-project="selectedProject"
+                            hide-create-project="true"
+                            hide-label="true">
+            </select-project>
+          </div>
+        </div>
+      </fieldset>
+    </ng-form>
+  </div>
+  <div class="modal-footer">
+      <button class="btn btn-lg btn-primary" type="button"
+              ng-disabled="(preferredHomePage === 'project-overview' && form.$invalid) || form.$pristine"
```

Comment:
It would be better to update `select-project` to take an optional required attribute that is set to true only when `preferredHomePage === 'project-overview'`. Then this changes to

```
ng-disabled="form.$invalid || form.$pristine"
```

Reply:
Hi @spadgett, sorry, I little confused on "...update select-project to take an optional required attribute..",  Can you please elaborate on 'optional required attribute`? -Thanks

Reply:
Sorry I phrased that poorly. I mean something like

```html
<select-project is-required="preferredHomePage === 'project-overview'" ...>
```

Then `form.$invalid` will be the correct value regardless of `preferredHomePage`